src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,247 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import {
+  deleteTodo,
+  getTodos,
+  updateTodoStatus,
+  updateTodoTitle,
+} from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem';
+import cn from 'classnames';
+
+type Props = {
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  setErrorMessage: React.Dispatch<React.SetStateAction<string>>;
+  filteredTodos: Todo[];
+  todos: Todo[];
+  tempTodo: Todo | null;
+  setLoadingTodosId: React.Dispatch<React.SetStateAction<number[]>>;
+  loadingTodosId: number[];
+  addIdToLoad: (arg: number) => void;
+};
+
+export const TodoList: React.FC<Props> = ({
+  setTodos,
+  setErrorMessage,
+  filteredTodos,
+  todos,
+  tempTodo,
+  setLoadingTodosId,
+  loadingTodosId,
+  addIdToLoad,
+}) => {
+  const [editingTodo, setEditingTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const cancelEditing = useCallback(() => {
+    setQuery('');
+    setLoadingTodosId([]);
+    setEditingTodo(null);
+  }, [setLoadingTodosId]);
+
+  useEffect(() => {
+    const handleKeyUp = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        event.preventDefault();
+        cancelEditing();
+      }
+    };
+
+    document.addEventListener('keyup', handleKeyUp);
+
+    return () => {
+      document.removeEventListener('keyup', handleKeyUp);
+    };
+  }, [cancelEditing]);
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      try {
+        const fetchedTodos = await getTodos();
+
+        setTodos(fetchedTodos);
+      } catch (error) {
+        setErrorMessage('Unable to load todos');
+        throw error;
```

Comment:
do not throw an error here - you handled it already, so no need to throw it again. It will lead to a page crash (because you don't try to catch the error thrown here somewhere else)